### PR TITLE
feat(vmd.logger): add warning level to core logging framework

### PR DESCRIPTION
**Pull request by kieronlanning**
Add warning level to enable clients to meaningfully representing warning-level log messages.

### CODE REVIEW

**Review by cedricbrasey — DISMISSED** [review]

**Review by mogusbi-motech — CHANGES_REQUESTED** [review]
I can't see any changes to `package.json` so could you undo the changes to the `package-lock.json` file please. This was probably because you ran `npm i` instead of `npm ci`.

Package locks should be frozen unless changes to dependencies have been made.

**Comment by kieronlanning** [comment]
@mogusbi-motech Updated.

File: src/logging/__tests__/vmd-logger.test.ts [hunk 1]
```diff
@@ -78,6 +79,24 @@ describe('VMD logged', () => {
     );
   });
 
+  it('should create an warn log when logRequestWarn is called', () => {
```
Comment (nroper) [inline src/logging/__tests__/vmd-logger.test.ts:82]: Should this be "... a warn log..." instead of "... an warn log..." ?

**Review by richmilton — DISMISSED** [review]
LGTM

File: src/logging/__tests__/vmd-logger.test.ts (hunk 1 above)
Comment (kieronlanning) [reply src/logging/__tests__/vmd-logger.test.ts:82]: You found my mad copy and paste skills the test above, I see... Up to you if you want that changed in this release or not.

File: src/logging/__tests__/vmd-logger.test.ts [hunk 2]
```diff
@@ -114,6 +133,15 @@ describe('VMD logged', () => {
     );
   });
 
+  it('should create an info warn when logWarn is called', () => {
```
Comment (mogusbi-motech) [inline src/logging/__tests__/vmd-logger.test.ts]: This should be `should create a warn log when logWarn is called`

File: src/logging/__tests__/vmd-logger.test.ts (hunk 1 above)
Comment (kieronlanning) [reply src/logging/__tests__/vmd-logger.test.ts]: Resolved.

File: src/logging/__tests__/vmd-logger.test.ts (hunk 2 above)
Comment (nroper) [inline src/logging/__tests__/vmd-logger.test.ts]: Also not sure if this is correct: 'info warn'

**Review by mogusbi-motech — APPROVED** [review]
Approved but it would be good to fix those test spec descriptions. We should always aim for the highest standards and this should apply to tests as well.

File: src/logging/__tests__/vmd-logger.test.ts (hunk 2 above)
Comment (kieronlanning) [reply src/logging/__tests__/vmd-logger.test.ts]: Updated too.

Comment (kieronlanning) [reply src/logging/__tests__/vmd-logger.test.ts]: Updated. already

File: src/logging/__tests__/vmd-logger.test.ts (hunk 1 above)
Comment (nroper) [reply src/logging/__tests__/vmd-logger.test.ts]: Guess it wouldn't hurt if you're making other changes re package-lock.json

**Review by nroper — APPROVED** [review]

**Review by richmilton — APPROVED** [review]
LGTM

**Pull request merged**